File: Test/Desktop/System.Common.DomainServices.Test/Shared/Global.shared.vb
' Used in codegen tests to test shared global types.

Imports System
Imports System.ComponentModel.DataAnnotations

Public Class GlobalNamespaceTest_Validation
    Public Shared Function Validate(ByVal input As String) As ValidationResult
        Return ValidationResult.Success
    End Function
End Class

Public Class GlobalNamespaceTest_ValidationAttribute
    Inherits ValidationAttribute
    Protected Overrides Function IsValid(ByVal value As Object, ByVal validationContext As ValidationContext) As ValidationResult
        Return ValidationResult.Success
    End Function
End Class

Public Class GlobalNamespaceTest_Attribute
    Inherits Attribute
    Public Property EnumProperty As GlobalNamespaceTest_Enum
End Class


Public Enum GlobalNamespaceTest_Enum
    DefaultValue
    NonDefaultValue
End Enum

File: Test/Desktop/System.Common.DomainServices.Test/Shared/Global.shared.cs
// Used in codegen tests to test shared global types.

using System;
using System.ComponentModel.DataAnnotations;

public static class GlobalNamespaceTest_Validation
{
    public static ValidationResult Validate(string input)
    {
        return ValidationResult.Success;
    }
}

public class GlobalNamespaceTest_ValidationAttribute : ValidationAttribute
{
    protected override ValidationResult IsValid(object value, ValidationContext validationContext)
    {
        return ValidationResult.Success;
    }
}

public class GlobalNamespaceTest_Attribute : Attribute
{
    public GlobalNamespaceTest_Enum EnumProperty { get; set; }
}

public enum GlobalNamespaceTest_Enum
{
    DefaultValue,
    NonDefaultValue,
}